x-pack/plugins/code/server/security.ts

```diff
@@ -61,14 +61,14 @@ export class SecureRoute {
     }
 
     this.server.securedRoute = securedRoute;
-  }
+}
 
   private isSecurityEnabledInEs() {
     const xpackInfo = this.server.plugins.xpack_main.info;
-    if (
-      xpackInfo.isAvailable() &&
-      (!xpackInfo.feature('security').isEnabled() || xpackInfo.license.isOneOf('basic'))
-    ) {
+    if (!xpackInfo.isAvailable()) {
+      throw Boom.methodNotAllowed('x-pack info is not available yet.');
+    }
+    if (!xpackInfo.feature('security').isEnabled() || xpackInfo.license.isOneOf('basic')) {
```

Comment:
This will need to be updated at some point to be `!xpackInfo.feature('security').isEnabled() || !xpackInfo.feature('security').isAvailable()`. It's fine if this will be addressed in a subsequent PR.